internal/account/service.go

```diff
@@ -47,11 +54,35 @@ func (a *AccountService) RegisterAccount(ctx context.Context, spec RegisterAccou
 	return a.accountDBAccessor.RegisterAccount(ctx, account)
 }
 
+func (a *AccountService) Login(ctx context.Context, spec AccountCredentialSpec) (string, error) {
+	// Find the account by email
+	account, err := a.accountDBAccessor.FindAccountByEmail(ctx, spec.Email)
+	if err != nil {
+		return "", fmt.Errorf("account not found: %w", err)
+	}
+
+	// Verify the password
+	if err := account.VerifyPassword(spec.Password); err != nil {
+		return "", fmt.Errorf("invalid password: %w", err)
+	}
```

Comment:
This is great. An alternative approach is to issue a generic error message such as 'invalid credentials.' This reduces the information available to attackers and making it harder for them to determine which specific credentials are incorrect

Reply:
I agree. In this case, providing less information, such as a generic 'invalid credentials' message, enhances security by making it harder for attackers to determine which specific credentials are incorrect. I'll update it soon

Reply:
Yap, returning the error message can be implemented if on logging system (which we don't currently implement). Current version is already fine though since you have changed it to generic message.